lib/engine/game/g_18_zoo/step/choose_ability_on_sr.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G18ZOO
+      module ChooseAbilityOnSr
+        private
+
+        def can_choose_ability?(company)
+          entity = @game.current_entity
+          return false unless entity.player?
+          return false unless company.owner == entity
+
+          return true if @game.zoo_ticket?(company)
+          return can_choose_midas?(entity) if company == @game.midas
+          return can_choose_holiday?(entity) if company == @game.holiday
+          return can_choose_whatsup?(entity) if company == @game.whatsup
+          return can_choose_greek?(entity) if company == @game.it_is_all_greek_to_me
+
+          false
+        end
+
+        def can_choose_any_ability?(entity)
+          return false unless entity.player?
+
+          return true if @game.zoo_tickets?(entity)
+          return can_choose_midas?(entity) if @game.midas.owner == entity
```

Comment:
Can one player own multiple abilities? If so, do you need to check those too before returning?

Reply:
Also from experience, I've found its better to have this type of logic in the game class instead of the steps. Something to consider as you develop the game.

Reply:
done in 2edf5ad23

Reply:
I didn't move into game class, because for this specific case I'm using them only a single Step.
for the choose_ability_on_or I'll move on game later, as you suggest, because I will need them for more than one step